worker-workflow/src/main/java/com/github/cafdataprocessing/TransformedWorkflowCacheKey.java

```diff
@@ -0,0 +1,47 @@
+package com.github.cafdataprocessing;
+
+import java.util.Objects;
+
+/**
+ * Represents a key to a transformed workflow cache entry
+ */
+final class TransformedWorkflowCacheKey {
+    private final String projectId;
+    private final long workflowId;
+
+    /**
+     * Create a transformed workflow cache key using the project ID and workflow ID provided.
+     * @param projectId project ID of the transformed workflow this key is to be associated with.
+     * @param workflowId workflow ID of the transformed workflow this key is to be associated with.
+     */
+    public TransformedWorkflowCacheKey(final String projectId, final long workflowId) {
+        this.projectId = projectId;
+        this.workflowId = workflowId;
+    }
+
+    public String getProjectId() {
+        return projectId;
+    }
+
+    public long getWorkflowId() {
+        return workflowId;
+    }
+
+    @Override
+    public boolean equals(final Object  o) {
+        if(this == o) {
+            return true;
+        }
+        if(o == null || getClass() != o.getClass()) {
```

Comment:
Using `if(o == null || getClass() != o.getClass())` is fine but because the class is `final` there is no possibility of any subclasses so you can use `if (!(o instanceof TransformedWorkflowCacheKey))` instead which I think is a little easier.